--- openvswitch-2.0.0/tests/automake.mk~	2013-10-16 00:50:35.000000000 +0200
+++ openvswitch-2.0.0/tests/automake.mk	2013-12-04 20:40:57.780242023 +0100
@@ -100,7 +100,6 @@
 	tests/valgrind/ovsdb-server \
 	tests/valgrind/ovsdb-tool \
 	tests/valgrind/test-aes128 \
-	tests/valgrind/test-atomic \
 	tests/valgrind/test-bundle \
 	tests/valgrind/test-byte-order \
 	tests/valgrind/test-classifier \
@@ -179,10 +178,6 @@
 tests_test_aes128_SOURCES = tests/test-aes128.c
 tests_test_aes128_LDADD = lib/libopenvswitch.a $(SSL_LIBS)
 
-noinst_PROGRAMS += tests/test-atomic
-tests_test_atomic_SOURCES = tests/test-atomic.c
-tests_test_atomic_LDADD = lib/libopenvswitch.a $(SSL_LIBS)
-
 noinst_PROGRAMS += tests/test-bundle
 tests_test_bundle_SOURCES = tests/test-bundle.c
 tests_test_bundle_LDADD = lib/libopenvswitch.a $(SSL_LIBS)
