--- openvswitch-1.4.0/utilities/ovs-ctl.in~	2012-01-31 08:08:18.000000000 +0100
+++ openvswitch-1.4.0/utilities/ovs-ctl.in	2012-03-24 20:56:17.184407921 +0100
@@ -33,9 +33,10 @@
 insert_openvswitch_mod_if_required () {
     # If openvswitch_mod is already loaded then we're done.
     test -e /sys/module/openvswitch_mod && return 0
+    test -e /sys/module/openvswitch && return 0
 
     # Load openvswitch_mod.  If that's successful then we're done.
-    action "Inserting openvswitch module" modprobe openvswitch_mod && return 0
+    action "Inserting openvswitch module" modprobe openvswitch_mod || modprobe openvswitch && return 0
 
     # If the bridge module is loaded, then that might be blocking
     # openvswitch_mod.  Try to unload it, if there are no bridges.
@@ -48,11 +49,12 @@
     action "removing bridge module" rmmod bridge || return 1
 
     # Try loading openvswitch_mod again.
-    action "Inserting openvswitch module" modprobe openvswitch_mod
+    action "Inserting openvswitch module" modprobe openvswitch_mod || modprobe openvswitch
 }
 
 insert_brcompat_mod_if_required () {
     test -e /sys/module/brcompat_mod && return 0
+    test -e /sys/module/openvswitch && return 0
     action "Inserting brcompat module" modprobe brcompat_mod
 }
 
